### PR TITLE
fix[hmi-server]: Download netcdf assets

### LINKED ISSUE #3206: [FEAT]: Allow saving of netcdf data

### PR DESCRIPTION
**Pull request by dvince2**
# Description
Adding support for downloading the original netcdf files from the source service. If there are download urls set this will take precidence over s3 stored assets, but we _should_ never have both set as subset datasets are new datasets without the dataset urls field set.

Resolves #3206

### CODE REVIEW

**Comment by dvince2** [comment]
If this isn't needed for the eval scenario then lets leave this out today. There's no point in rocking the boat?

File: packages/client/hmi-client/src/types/Types.ts [hunk 1]
```diff
@@ -160,6 +160,7 @@ export interface Dataset extends TerariumAsset {
     dataSourceDate?: Date;
     fileNames?: string[];
     datasetUrl?: string;
+    datasetUrls?: string[];
```
Comment (YohannParis) [inline packages/client/hmi-client/src/types/Types.ts:163]: Why do we need both?

Comment (dvince2) [reply packages/client/hmi-client/src/types/Types.ts:163]: We don't, but I'm only about 80% sure we don't. `datasetUrl` is a holdover from old TDS. in `Dataset.java` I've added a comment with a link to a github issue to confirm this and remove after the eval - I dont want to commit that and have it blow up

Comment (bigglesandginger) [reply packages/client/hmi-client/src/types/Types.ts:163]: I agree that it is most probably a holdover, but my knowledge is less with respect to this.  And I love the idea that during the eval, maybe not the time to delete it :)

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java [hunk 2]
```diff
@@ -375,17 +376,54 @@ public ResponseEntity<StreamingResponseBody> getFile(
...
+		} catch (final IOException e) {
+			final String error = "Unable to get dataset";
 			log.error(error, e);
 			throw new ResponseStatusException(
 					org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
 					error);
 		}
+
+		if(dataset.get().getEsgfId() != null && !dataset.get().getEsgfId().isEmpty()
+			&& dataset.get().getDatasetUrls() != null && !dataset.get().getDatasetUrls().isEmpty()){
```
Comment (bigglesandginger) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java:396]: Liking - super solid

**Review by bigglesandginger — APPROVED** [review]

File: packages/client/hmi-client/src/types/Types.ts (hunk 1 above)
Comment (dvince2) [reply packages/client/hmi-client/src/types/Types.ts:163]: yes hahah :D

on the java model:
TODO: IS THIS NEEDED? IS THIS FROM OLD TDS? https://github.com/DARPA-ASKEM/terarium/issues/3194

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java (hunk 2 above)
Comment (dvince2) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java:396]: In general I prefer java to javascript... but I do love the brevity of just checking `if(!someVar)` and that does a null and empty check at the same time...

**Pull request merged**